### PR TITLE
rpc: replace reth-primitive imports

### PR DESCRIPTION
**Pull request by estensen**
Progress on https://github.com/paradigmxyz/reth/issues/10749

Changed all where direct drop-in-replacement was possible

### CODE REVIEW

File: crates/rpc/rpc-api/src/engine.rs [hunk 1]
```diff
@@ -4,9 +4,10 @@
 //! the consensus client.
 
 use alloy_json_rpc::RpcObject;
+use alloy_primitives::{Address, BlockHash, Bytes, B256, U256, U64};
 use jsonrpsee::{core::RpcResult, proc_macros::rpc};
 use reth_engine_primitives::EngineTypes;
-use reth_primitives::{Address, BlockHash, BlockId, BlockNumberOrTag, Bytes, B256, U256, U64};
+use reth_primitives::{BlockId, BlockNumberOrTag};
```
Comment (Rjected) [inline crates/rpc/rpc-api/src/engine.rs:10]: This could be an `alloy_eips` import

**Review by Rjected — DISMISSED** [review]
just have one suggestion for moving `BlockId` and `BlockNumberOrTag`, since we just re-export them in primitives

File: crates/rpc/rpc-api/src/engine.rs (hunk 1 above)
Comment (mattsse) [reply crates/rpc/rpc-api/src/engine.rs:10]: yeah, but we can do this separately

**Review by mattsse — APPROVED** [review]
followup for alloy_eips imports

**Pull request merged**